File: 2 - EP/python/cifra_cesar.py
import sys

alfabetoDefault = ['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y', 'z']

alfabetoCesar = ['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y', 'z']

def reCrypt(comando, mensagem, rotate, cryptLevel):
    
    if comando == "encrypt":
        for i in range(0,cryptLevel):
            rotateAlpha(rotate)
            mensagem = encrypt(mensagem)
            
        print(mensagem)

    elif comando == "decrypt":
        for j in range(0,cryptLevel):  
            rotateAlpha(rotate)
            mensagem = decrypt(mensagem)
    
        print(mensagem)

def rotateAlpha(rotate):

    countPositivo = 26 - rotate
    for p in range(0,rotate):
        alfabetoCesar[countPositivo] = alfabetoDefault[p]
        countPositivo = countPositivo + 1

    countNegativo = - 26 + rotate
    countReplace = 26 - rotate
    for t in range(0,countReplace):
        alfabetoCesar[t] = alfabetoDefault[countNegativo]
        countNegativo = countNegativo + 1
    
def decrypt(mensagem):
    
    msg = ''
    mensagemFormated = (mensagem.lower())

    for j in range(len(mensagemFormated)):
        if((mensagemFormated[j] in alfabetoDefault) == True):
            k = alfabetoCesar.index(mensagemFormated[j])
            msg += alfabetoDefault[k]

        elif((mensagemFormated[j] in alfabetoDefault) == False):
            msg += mensagemFormated[j]

    return msg

def encrypt(mensagem):
    
    msg = ''
    mensagemFormated = (mensagem.lower())

    for i in range(len(mensagemFormated)):
        if((mensagemFormated[i] in alfabetoDefault) == False):
            msg += mensagemFormated[i]

        elif((mensagemFormated[i] in alfabetoDefault) == True):
            j = alfabetoDefault.index(mensagemFormated[i])
            msg += alfabetoCesar[j]

    return msg

def main():

    comando = sys.argv[1]
    mensagem = sys.argv[2]
    rotate = int(sys.argv[3])
    cryptLevel = int(sys.argv[4])     

    rotateAlpha(rotate)

    if comando == "encrypt" or  comando == "decrypt":
        reCrypt(comando, mensagem, rotate, cryptLevel)
    else:
        print ('Comando invalido!!!')
    
if __name__ == '__main__':
    main()

File: 2 - EP/cesar_cifra/javascript/cifra_cesar.js
alphaDefault = new Array('a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z');
alphaCrypted = new Array('a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z');

var rotate = 0;
var cryptLevel = 1;

function reCrypt(comando, mensagem, rotate, cryptLevel) {

    if (comando == "encrypt") {
        for (var i = 0; i < cryptLevel; i++) {
            rotateAlpha(rotate);
            mensagem = encrypt(mensagem);
        }
    } else if (comando == "decrypt") {
        for (var i = 0; i < cryptLevel; i++) {
            rotateAlpha(rotate);
            mensagem = decrypt(mensagem);
        }
    }

    return mensagem;
}

function rotateAlpha(rotate) {
    var countPositivo = 26 - rotate;
    for (var i = 0 & countPositivo; i < rotate; i++ & countPositivo++) {
        alphaCrypted[countPositivo] = alphaDefault[i];
    }

    j = rotate;
    for (var i = 0 & j; j < 26; i++ & j++) {
        alphaCrypted[i] = alphaDefault[j];
        //console.log('i->', i, 'j->', j);
    }
}

function decrypt(mensagem) {

    msgEncrypting = "";
    mensagemFormated = (mensagem.toLowerCase());

    for (var i = 0; i < mensagemFormated.length; i++) {
        if ((alphaDefault.indexOf(mensagemFormated[i]) != -1)) {
            var j = alphaCrypted.indexOf(mensagemFormated[i]);
            msgEncrypting += alphaDefault[j];
        } else {
            msgEncrypting += mensagemFormated[i];
        }
    }
    return msgEncrypting;
}

function encrypt(mensagem) {
    msgEncrypting = [];
    mensagemFormated = (mensagem.toLowerCase());

    for (var i = 0; i < mensagemFormated.length; i++) {
        if ((alphaDefault.indexOf(mensagemFormated[i]) != -1)) {
            var j = alphaDefault.indexOf(mensagemFormated[i]);
            msgEncrypting += alphaCrypted[j];
        } else {
            msgEncrypting += mensagemFormated[i];
        }
    }
    return msgEncrypting;
}

function msgEncrypted(txtInput) {

    var msgCriptograda = reCrypt("encrypt", txtInput.value + '', rotate, cryptLevel);

    return msgCriptograda;
}

function msgDescrypted(txtInput) {

    var msgDescriptografada = reCrypt("decrypt", txtInput.value + '', rotate, cryptLevel);

    return msgDescriptografada;

}

function main() {


    var btnEncrypt = document.getElementById("btnEncrypt");
    var btnDecrypt = document.getElementById("btnDecrypt");

    var btnRotateAdd = document.getElementById("btnRotateAdd");
    var btnRotateDec = document.getElementById("btnRotateDec");
    var btnCryptLevelAdd = document.getElementById("btnCryptLevelAdd");
    var btnCryptLevelDec = document.getElementById("btnCryptLevelDec");

    var txtInput = document.getElementById("txtInput");
    var txtOutput = document.getElementById("txtOutput");

    var rotateValue = document.getElementById('rotateValue');

    rotateCryptLevel.setAttribute("data-text", cryptLevel + '');

    btnEncrypt.onclick = function() {

        rotateAlpha(rotate);

        var msgCriptografada = msgEncrypted(txtInput);
        txtOutput.value = msgCriptografada;
    };

    btnDecrypt.onclick = function() {

        var msgDescriptografada = msgDescrypted(txtInput);
        txtOutput.value = msgDescriptografada;
    };

    btnRotateAdd.onclick = function() {

        if (rotate >= 0 && rotate < 13) {
            rotate++;
            rotateValue.setAttribute('data-text', rotate + '');
        }

        var msgCriptografada = msgEncrypted(txtInput);
        txtOutput.value = msgCriptografada;
    };

    btnRotateDec.onclick = function() {

        if (rotate > 0 && rotate <= 13) {
            rotate--;
            rotateValue.setAttribute("data-text", rotate + '');
        }

        var msgCriptografada = msgEncrypted(txtInput);
        txtOutput.value = msgCriptografada;
    };

    btnCryptLevelAdd.onclick = function() {

        if (cryptLevel <= 0 && cryptLevel > 13) {
            cryptLevel++;
            rotateCryptLevel.setAttribute("data-text", cryptLevel + '');
        }
        var msgCriptograda = msgEncrypted(txtInput);
        txtOutput.value = msgCriptograda;

        var msgDescriptografada = msgDescrypted(txtInput);

    };

    btnCryptLevelDec.onclick = function() {

        if (cryptLevel > 0) {
            cryptLevel--;
            rotateCryptLevel.setAttribute("data-text", cryptLevel + '');
        }

        var msgCriptograda = msgEncrypted(txtInput);
        txtOutput.value = msgCriptograda;

    };
}